applications/sdp/mspi/src/main.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "./hrt/hrt.h"
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <hal/nrf_vpr_csr.h>
+#include <hal/nrf_vpr_csr_vio.h>
+#include <haly/nrfy_gpio.h>
+
+#include <zephyr/drivers/mspi.h>
+
+#define MAX_DATA_LEN 256
+
+#define XFER_COMMAND_IDX (0)
+#define XFER_ADDRESS_IDX (1)
+#define XFER_DATA_IDX	 (2)
+
+#define HRT_IRQ_PRIORITY	   2
+#define HRT_VEVIF_IDX_WRITE_SINGLE 17
+#define HRT_VEVIF_IDX_WRITE_QUAD   18
+
+/* how many words are needed for given amount of bytes.*/
+#define WORDS_FOR_BYTES(bytes) ((bytes - 1) / 4 + 1)
+
+#define VEVIF_IRQN(vevif)   VEVIF_IRQN_1(vevif)
+#define VEVIF_IRQN_1(vevif) VPRCLIC_##vevif##_IRQn
+
+struct mspi_config {
+	uint8_t *data;
+	uint8_t data_len;
+	uint8_t word_size;
+};
+
+struct mspi_dev_config {
+	enum mspi_io_mode io_mode;
+	enum mspi_ce_polarity ce_polarity;
+	uint32_t read_cmd;
+	uint32_t write_cmd;
+	uint8_t cmd_length;  /* Command length in bits. */
+	uint8_t addr_length; /* Address length in bits. */
+};
+
+static struct mspi_dev_config mspi_dev_configs;
+
+uint32_t data_buffer[MAX_DATA_LEN + 2];
+
+volatile struct hrt_ll_xfer xfer_ll_params = {.counter_top = 4,
+					      .word_size = 4,
+					      .data_to_send = NULL,
+					      .data_len = 0,
+					      .ce_hold = false,
+					      .ce_enable_state = false};
+
+void configure_clock(enum mspi_cpp_mode cpp_mode)
+{
+	nrf_vpr_csr_vio_config_t vio_config = {
+		.input_sel = 0,
+		.stop_cnt = 0,
+	};
```

Comment:
match styles i.e. using bottom, { then newline then elements, one per line, then newline for };

Reply:
I'm not changing that, this is how clang-format does it. If I would change it I will have to remember about it every time I use clang-format

Reply:
It is like that because the top struct is missing a comma after the final element, add it and it will format correctly

Reply:
ok, thanks, fixed